### PR TITLE
Feat/categories section component

### PR DESCRIPTION
**Pull request by tranchinh123**

### CODE REVIEW

File: bookshelter-app/src/components/CategoriesSection/index.tsx [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
+const CategoriesSection = ({ currentCategory }: CategoriesSectionProps) => {
   return (
     <section className={styles.CategoriesSection}>
-      <p>Categories</p>
-      <div>
-        <img src={arrow} alt="" />
-        <p>Showing 6 Result(s)</p>
-      </div>
+      <p className={styles.categories}>Categories</p>
+      {currentCategory ? (
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/CategoriesSection/index.tsx]: use `condition && ...` syntax instead

File: bookshelter-app/src/components/CategoriesSection/index.tsx [hunk 2]
```diff
@@ -1,13 +1,22 @@
...
+        <div className={styles.wrapped}>
+          <div className={styles.category}>{currentCategory}</div>
+          <img src={arrow} alt="" />
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/CategoriesSection/index.tsx]: use SVG for icon instead

File: bookshelter-app/src/components/CategoriesSection/index.tsx [hunk 3]
```diff
@@ -1,13 +1,22 @@
...
+          <p>Showing 6 Result(s)</p>
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/CategoriesSection/index.tsx]: define prop for results

File: bookshelter-app/src/components/CategoriesSection/index.tsx (hunk 1 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/CategoriesSection/index.tsx]: Fixed!

File: bookshelter-app/src/components/CategoryList/index.tsx [hunk 4]
```diff
@@ -1,39 +1,33 @@
-import { useEffect, useState } from 'react';
 import styles from './index.module.css';
 import BookCategory from '../BookCategory';
-import { get } from '../../services/api';
-import { API } from '../../constants/api';
-import { Categories } from '../../types';
 import getRandomColor from '../../utils/randomColor';
+interface Category {
```
Comment (huypham-agilityio) [inline bookshelter-app/src/components/CategoryList/index.tsx]: move to types folder

File: bookshelter-app/src/layouts/index.tsx [hunk 5]
```diff
@@ -2,16 +2,39 @@ import styles from './index.module.css';
...
+
   return (
     <>
       <Header />
-      <CategoriesSection />
-      <CategoryList />
+      <CategoriesSection currentCategory={currentCategory} />
+      <CategoryList
+        categories={categories}
+        onCategoryClick={handleCategoryClick}
```
Comment (huypham-agilityio) [inline bookshelter-app/src/layouts/index.tsx]: onClick is enough, similar update for other places

File: bookshelter-app/src/layouts/index.tsx [hunk 6]
```diff
@@ -2,16 +2,39 @@ import styles from './index.module.css';
...
   children: React.ReactNode;
 }
+
 const DefaultLayout = ({ children }: LayoutProps) => {
+  const [categories, setCategories] = useState<Categories[]>([]);
+  const [currentCategory, setCurrentCategory] = useState<string | null>(null);
+  
+  useEffect(() => {
+    const fetchCategoriesList = async (): Promise<void> => {
+      const Categories = await get<Categories[]>(API.CATEGORIES_ENDPOINT);
```
Comment (huypham-agilityio) [inline bookshelter-app/src/layouts/index.tsx]: lowercase

File: bookshelter-app/src/components/CategoriesSection/index.tsx (hunk 2 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/CategoriesSection/index.tsx]: Fixed!

File: bookshelter-app/src/layouts/index.tsx (hunk 5 above)
Comment (tranchinh123) [reply bookshelter-app/src/layouts/index.tsx]: Fixed!

File: bookshelter-app/src/layouts/index.tsx (hunk 6 above)
Comment (tranchinh123) [reply bookshelter-app/src/layouts/index.tsx]: Fixed!

File: bookshelter-app/src/components/CategoryList/index.tsx (hunk 4 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/CategoryList/index.tsx]: Fixed!

File: bookshelter-app/src/components/CategoriesSection/index.tsx (hunk 3 above)
Comment (tranchinh123) [reply bookshelter-app/src/components/CategoriesSection/index.tsx]: Fixed!

**Pull request merged**